struqture-py-macros/src/mappings.rs

```diff
@@ -0,0 +1,156 @@
+            output_wrapper_type = quote::format_ident!("{}", output_wrapper_name);
+
+            quote! {
+                /// Transform the given spin object into a fermionic object using
+                /// the Jordan Wigner mapping.
+                pub fn jordan_wigner(&self) -> #output_wrapper_type {
+                    #output_wrapper_type {
+                        internal: self.internal.jordan_wigner()
+                    }
+                }
+            }
+        } else {
+            if struct_name == "PauliProduct"
+                || struct_name == "DecoherenceProduct"
+                || struct_name == "PlusMinusProduct"
+                || struct_name == "PlusMinusOperator"
+            {
+                output_wrapper_type = quote::format_ident!("FermionSystemWrapper");
+                output_type = quote::format_ident!("FermionSystem");
+            } else if struct_name == "PlusMinusLindbladNoiseOperator" {
+                output_wrapper_type = quote::format_ident!("FermionLindbladNoiseSystemWrapper");
+                output_type = quote::format_ident!("FermionLindbladNoiseSystem");
+            } else {
+                panic!("JordanWignerSpinToFermion can only be implemented for spin types!")
+            };
+
+            quote! {
+                /// Transform the given spin object into a fermionic object using
+                /// the Jordan Wigner mapping.
+                pub fn jordan_wigner(&self) -> #output_wrapper_type {
+                    #output_wrapper_type {
+                        internal: #output_type::from_operator(
+                            self.internal.jordan_wigner(), None)
+                            .expect("Internal bug when creating fermionic system object from fermionic operator object.")
+                    }
+                }
+            }
+        }
+    } else {
+        TokenStream::new()
+    }
+}
+
+fn jordan_wigner_fermion_to_spin_quote(
+    attribute_arguments: &AttributeMacroArguments,
+    struct_name: &str,
+) -> TokenStream {
+    if attribute_arguments.contains("JordanWignerFermionToSpin") {
+        let output_wrapper_type;
+        let output_type;
+        let from_method;
+
+        if struct_name.contains("System") {
+            let mut output_wrapper_name = format!("{}Wrapper", struct_name);
+            output_wrapper_name = output_wrapper_name.replace("Fermion", "Spin");
+            output_wrapper_type = quote::format_ident!("{}", output_wrapper_name);
+
+            quote! {
+                /// Transform the given fermionic object into a spin object using
+                /// the Jordan Wigner mapping.
+                pub fn jordan_wigner(&self) -> #output_wrapper_type {
+                    #output_wrapper_type {
+                        internal: self.internal.jordan_wigner()
+                    }
+                }
+            }
+        } else {
+            if struct_name == "FermionProduct" {
```

Comment:
How come you use `==` here and `.contains` above? Should this be the same in both places? (also same comment for the function above)

Reply:
The output type for all fermion `System` types can be built automatically (it is just the corresponding spin `System` type, see lines 115-117), while for the other spin types we have to specify it manually one by one.